test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc

```diff
@@ -175,6 +175,116 @@ TEST_P(GrpcJsonTranscoderIntegrationTest, UnaryPost) {
+  // Using the following HTTP template:
+  //   POST /shelves
+  //   body: shelf
+  testTranscoding<bookstore::CreateShelfRequest, bookstore::Shelf>(
+      Http::TestHeaderMapImpl{{":method", "POST"},
+                              {":path", "/shelf?shelf.theme=Children"},
+                              {":authority", "host"},
+                              {"content-type", "application/json"}},
+      "", {R"(shelf { theme: "Children" })"}, {R"(id: 20 theme: "Children" )"}, Status(),
+      Http::TestHeaderMapImpl{
+          {":status", "200"},
+          {"content-type", "application/json"},
+      },
+      R"({"id":"20","theme":"Children"})");
+
+  // 2. Binding theme='Children' and id='999' in CreateShelfRequest
+  testTranscoding<bookstore::CreateShelfRequest, bookstore::Shelf>(
+      Http::TestHeaderMapImpl{{":method", "POST"},
+                              {":path", "/shelf?shelf.id=999&shelf.theme=Children"},
+                              {":authority", "host"},
+                              {"content-type", "application/json"}},
+      "", {R"(shelf { id: 999 theme: "Children" })"}, {R"(id: 999 theme: "Children" )"}, Status(),
+      Http::TestHeaderMapImpl{
+          {":status", "200"},
+          {"content-type", "application/json"},
+      },
+      R"({"id":"999","theme":"Children"})");
+
+  // 3. Binding shelf=1, book=<post body> and book.title='War and Peace' in CreateBookRequest
+  //    Using the following HTTP template:
+  //      POST /shelves/{shelf}/books
+  //      body: book
+  testTranscoding<bookstore::CreateBookRequest, bookstore::Book>(
+      Http::TestHeaderMapImpl{{":method", "PUT"},
+                              {":path", "/shelves/1/books?book.title=War%20and%20Peace"},
+                              {":authority", "host"}},
+      R"({"author" : "Leo Tolstoy"})",
+      {R"(shelf: 1 book { author: "Leo Tolstoy" title: "War and Peace" })"},
+      {R"(id: 3 author: "Leo Tolstoy" title: "War and Peace")"}, Status(),
+      Http::TestHeaderMapImpl{{":status", "200"}, {"content-type", "application/json"}},
+      R"({"id":"3","author":"Leo Tolstoy","title":"War and Peace"})");
+
+  // 4. Binding shelf=1, book.author='Leo Tolstoy' and book.title='War and Peace' in
+  // CreateBookRequest
+  //    Using the following HTTP template:
+  //      POST /shelves/{shelf}/books
+  //      body: book
+  testTranscoding<bookstore::CreateBookRequest, bookstore::Book>(
+      Http::TestHeaderMapImpl{
+          {":method", "PUT"},
+          {":path", "/shelves/1/books?book.author=Leo%20Tolstoy&book.title=War%20and%20Peace"},
+          {":authority", "host"}},
+      "", {R"(shelf: 1 book { author: "Leo Tolstoy" title: "War and Peace" })"},
+      {R"(id: 3 author: "Leo Tolstoy" title: "War and Peace")"}, Status(),
+      Http::TestHeaderMapImpl{{":status", "200"}, {"content-type", "application/json"}},
+      R"({"id":"3","author":"Leo Tolstoy","title":"War and Peace"})");
+
+  // 5. Test URL decoding.
```

Comment:
this case is covered by 3) and 4),

Reply:
LGTM

Reply:
This test is for url encoding.

Reply:
But both test 3 and 4 have url encoding already with "War and Peace" encoded.